rclc/CMakeLists.txt

```diff
@@ -1,56 +1,126 @@
 cmake_minimum_required(VERSION 3.5)
+set(CMAKE_C_CLANG_TIDY clang-tidy -checks=*)
 
 project(rclc)
 
+#################################################
+# compiler settings
+#################################################
+
+# Default to C11
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 11)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+# for performance tracing
+# see https://gitlab.com/micro-ROS/ros_tracing/ros2_perf_analysis/tree/feature/simple-analysis/scripts
+#add_compile_options(-finstrument-functions
+#-fplugin=~/ros_ws/src/instrument-attribute-gcc-plugin/instrument_attribute.so
+#-fplugin-arg-instrument_attribute-debug)
+
+set(CMAKE_VERBOSE_MAKEFILE ON)
+
+#################################################
+# package dependencies
+#################################################
 find_package(ament_cmake_ros REQUIRED)
 find_package(rcl REQUIRED)
+find_package(rcutils REQUIRED)
+find_package(std_msgs REQUIRED)
 find_package(rosidl_generator_c REQUIRED)
 
-ament_export_include_directories(include)
-include_directories(include)
 
-if(NOT CMAKE_C_STANDARD)
-  set(CMAKE_C_STANDARD 11)
-endif()
+#################################################
+# include directories
+#################################################
+include_directories(include)
 
-set(rclc_sources
-  "src/functions.c"
-)
+#################################################
+# create library
+#################################################
+add_library(${PROJECT_NAME} src/init.c src/publisher.c src/subscription.c src/timer.c src/node.c src/executor_handle.c src/executor.c)
 
-add_library(${PROJECT_NAME}
-  SHARED
-  ${rclc_sources}
+target_include_directories(${PROJECT_NAME}
+    PUBLIC
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
 )
 
 ament_target_dependencies(${PROJECT_NAME}
   rcl
+  rcutils
   rosidl_generator_c
 )
 
-ament_export_dependencies(
-  rcl
-  rmw
-  rosidl_generator_c
-)
+
+#################################################
+# install
+#################################################
+
+install(DIRECTORY include/ DESTINATION include)
+install(TARGETS ${PROJECT_NAME} DESTINATION lib)
+
+# specific order: dependents before dependencies
 ament_export_include_directories(include)
 ament_export_libraries(${PROJECT_NAME})
 
+#################################################
+# gtest
+#################################################
+
```

Comment:
```suggestion
```

Reply:
fixed in 54f0bbc54a33cc601764ab1d53fe3119d9e61c8b